activity_browser/logger.py

```diff
@@ -0,0 +1,98 @@
+from PySide2.QtCore import Slot
+import logging
+import os, time
+import random, string
+
+
+from .signals import signals
+class Logger(logging.Logger):
+    """
+    Creates two logging streams, one to the console (with >=INFORMATION logs)
+    the second to a file specified with the name argument (with >=WARNING logs)
+    and attaches these to a logger.
+
+    Returns the logging device
+    """
+    file_handler = None
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.setLevel(logging.INFO)
+
+        self.stream_handler = logging.StreamHandler()
+        self.stream_handler.setLevel(logging.INFO)
+        # Threads have their own Logger, but these refer to the sessions log file
+        if Logger.file_handler is None:
+            dir_path = os.getcwd() + "/.logs"
+            os.makedirs(dir_path, exist_ok=True)
+            Logger.cleanDirectory(dir_path)
+            name = name + "_" + Logger.uniqueString(8) + '.log'
+            file_path = dir_path + "/" + name
+            Logger.file_handler = logging.FileHandler(file_path)
+        Logger.file_handler.setLevel(logging.WARNING)
+
+        self.log_format = logging.Formatter("%(module)s - %(levelname)s - %(asctime)s - %(message)s")
```

Comment:
Oh thank god!!! I was struggling to keep this format ab wide and keep the module name. This is because I'm wrapping the class so that the previous use of the print methods can continue. I should probably have just forced a stricter way of using the logging.xxx, but still I'd prefer to only change what I'm actually working on. Module does now provide the module where the error/warning/info/debug log occurs